src/l500/l500-depth.cpp

```diff
@@ -297,6 +270,13 @@ namespace librealsense
         });
     }
 
+    rs2_sensor_mode get_resolution_from_width_height(int width, int height)
+    {
+        if (width == 640 || height == 640)
+            return RS2_SENSOR_MODE_VGA;
+        return RS2_SENSOR_MODE_XGA;
```

Comment:
For refactoring improve input verification, i.e. if user calls this with (-1,-4444) returning RS2_SENSOR_MODE_XGA is probably not correct.